코드 리뷰

senxation opened the pull request:
간단하게 코드 리뷰 한번 해봐줄려고 `master` <- `dev` 로 머지하는 pr 생성했습니다.
원래는 `dev` <- `feature/*` 로 하는건데 이미 `dev` 다 머지해버렸으니 경험해보는데 의미를 두면 좋겠네요.

수정사항은 `dev`에 반영하면 되고요.

flyingSquirrel-dev committed:
delete useless folder(img)

flyingSquirrel-dev committed:
delete useless folder(img)

flyingSquirrel-dev committed:
modify NearbyList UI

JEONGJIHUN committed:
installed npm

vjbhang committed:
Merge branch 'dev' of https://github.com/hogangnono/codestates into dev

Commit:
Make parent class of shape
*Other shape's will inheritance Shape class

Commit:
Make Circle and Rect

flyingSquirrel-dev committed:
modify MyDrawings and LoginModal

flyingSquirrel-dev committed:
add condition to check if user is loged in when clicking save button

vjbhang committed:
working on button logic..

Commit:
Make Line class
* Make two path when user click
* One path is generate according to user mouse move
* It only make only one line and can't extend

JEONGJIHUN committed:
made component each toggle function and moved functions to app.js modifying server load api

JEONGJIHUN committed:
Merge pull request #53 from cloiepark/feature/handlLogin

(modify) feature/handleLogin

JEONGJIHUN committed:
Merged latest file

JEONGJIHUN committed:
Merge pull request #55 from JEONGJIHUN/feature/mergedLatestFile

Feature/merged latest file

Commit:
Make polyline!!
* start with left click and end with right click
* animated when draw line
* Can't caculate zoom in out yet

JEONGJIHUN committed:
loaded categories, modifying add event listener

JEONGJIHUN committed:
Mergerd #57

JEONGJIHUN committed:
Merged dohee code

JEONGJIHUN committed:
Merge pull request #58 from JEONGJIHUN/feature/loadApiModification

Feature/load api modification

flyingSquirrel-dev committed:
leave a commit before pull upstream dev

JEONGJIHUN committed:
Modified div to checkbox using less

vjbhang committed:
from toggle to selection

Changing logic from toggle to selection
- state is reduced from amount of buttons to just one state that is
based on which buttoin is selected.

JEONGJIHUN committed:
Modified drawing button position

JEONGJIHUN committed:
Modified filter button using less and modifying addEveventListener

flyingSquirrel-dev committed:
modify My/Drawing(그리기) js file

vjbhang committed:
Added Color Toggling; Fixed Multiple Listeners

- Removed previous toggleButtonState listeners for each button
- Added 3 state -> shapes, selectedButton, loadedListener
  - shapes: gives each button a unique name
  - selectedButton: indicates the currently selected button
  - loadedListener: holds the listener of shape for removal before
adding new listener
- Renders buttons dynamically using map

vjbhang committed:
Merge pull request #7 from vjbhang/selectButtonDev

Added Select Button and color Toggle

vjbhang committed:
Merge branch 'subDev' of https://github.com/vjbhang/codestates into dev

vjbhang committed:
Merge pull request #8 from vjbhang/selectButtonDev

Added Color Toggling; Fixed Multiple Listeners

vjbhang committed:
After Shape Completion color toggle

- Color toggle for button after shape creation.

vjbhang committed:
Made it so Overlays will be selected

- Overlays will be selected based on button selection
  - was testing only for circle on all circumstances.
  - change value of overlays array to concerned index.

vjbhang committed:
Merge branch 'dev' of https://github.com/hogangnono/codestates into dev

vjbhang committed:
Merge etc..

flyingSquirrel-dev committed:
modifying deactive drawing/filter mode

vjbhang committed:
updated for line shape

- updated so that line shape works for restructured code.

Commit:
Draw circle interactivly
* draw circle using 'path' instead of 'ellipse'
* start with left click and end with left click

flyingSquirrel-dev committed:
modify Filter/Drawing mode, modify feature handling login

JEONGJIHUN committed:
Merge pull request #60 from vjbhang/dev

Completed Toggle Button(fixed from Toggle to Selection); Added isInShapeCreateMode state; Overlay Selection Refactoring

Commit:
Modify Circle is dissaper when zoom in zoom out
* using line Data just like polyline
* Not change ratio accordingly map

senxation commented:
기본적으로 몇가지 안되어있는 것들.

- client 쪽에 .eslint 룰 설정
    - src 아래에 둔건 그 아래에 파일들만 적용하려는 건지?
    - 파일들의 indent 룰이 없어서 산만함. indent가 space든 tab이든 몇개든 상관없지만 통일하는게 좋아요. 
    - js/css/less 간에 indent룰이 달라도 좋지만 같은 파일은 동일한 룰로
- 폴더/파일 네이밍 룰도 마찬가지에요.
    - camelCase 쓰는 경우와 PascalCase 쓰는 경우가 혼재하는데, react class에만 PascalCase를 쓴다든지 하는 정의가 필요함.

senxation commented on packages/client/src/CustomOverlay/Circle.less:
```diff
@@ -0,0 +1,11 @@
+ellipse {
```
file 위치를 `src/less`에 두는게 맞을듯.

senxation commented on packages/client/src/Module/Button.js:
```diff
@@ -0,0 +1,71 @@
+    FaArrowLeft
+} from 'react-icons/fa';
+import PropTypes from 'prop-types';
+import '../less/Drawing.less';
```
`less`를 import해서 쓰는 방식은 chrome 디버거를 통해 어느 파일에서 스타일을 정의했는지 찾기 어렵게 때문에 webpack의 less-loader를 사용해서 sourcemap을 만들어서 쓰는 방식이 더 좋아요. 이건 당장은 고치지 않아도 좋지만 참고

Commit:
Make inheritanvce relation ship between Shape.js and all other figures
* All of Figure will use 'path'

Commit:
Make 'TwoClickShape.js' that parent of circle and rect

senxation commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,307 @@
+        super(props);
+        this.bound = undefined;
+        this.drawList = {};
+        this.newToggleBox = {
```
상수로 빼놓는게 좋을듯. 
`server/categories.js`에도 정의되어있고 server나 client나 같이 쓰게 될텐데 어떻게 처리할지 정하는게 좋을듯

senxation commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,307 @@
+        this.setState({ map });
+        this.bound = map.getBounds();
+        this.mainPageLoad(map);
+        naver.maps.Event.addListener(map, 'idle', e => {
```
`idle` 이벤트가 브라우저 창 크기를 변경할때 너무 자주 발생하므로 debounce 처리하는게 좋겠음

senxation commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,307 @@
+        this.setState({ name: username });
+    };
+
+    mapOption = () => {
```
입력값도 없고, 리턴값도 고정값인데 함수로 만들어서 반환할 이유가 없어보임.

flyingSquirrel-dev replied:
자식인 LoginModal.js에서 username(string)으로 받아서 부모인 App.js의 this.state.name으로 반영하도록 수정해서 콜백함수로 사용하고 있습니다! PR 요청한 코드에 반영해두었습니다 :)

senxation replied:
@cloiepark mapOption 함수를 얘기한거였어요.

Commit:
Delete onRemove function
* It just inheritance method

senxation commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,307 @@
+        naver.maps.Event.addListener(map, 'idle', e => {
+            this.bound = map.getBounds();
+            this.mainPageLoad(map);
+            this.DataDelete();
```
`mainPageLoad`에서 데이터 로딩이 끝나는 시점에 맞게 기존핀을 제거하는 `DataDelete`를 처리해야할듯 함.

senxation commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,307 @@
+                            });
+                            overlay.setMap(map);
+                            this.drawList[el.id] = overlay;
+                        }
```
이후에 

```suggestion
                        } else {
                            this.drawList[el.id].setMap(null);
                            delete this.drawList[el.id];
                        }
                        
```
이렇게 해야하지 않을까?

JEONGJIHUN replied:
넵 해당 부분 수정했습니다.

senxation commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,307 @@
+                }
+            })
+            .catch(error => {
+                // if (error.response.status === 500) {
```
필요없음 제거

JEONGJIHUN replied:
해당 코드은 필요한 부분으로 사용하는 걸로 수정했습니다.

senxation commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,307 @@
+            });
+    };
+
+    DataDelete = () => {
```
갑자기 왜 대문자로 시작할까요?
이름도 `deleteDraw` / `unsetDraw` 등으로 해야할 듯

JEONGJIHUN replied:
넵 수정 완료했습니다.

senxation commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,307 @@
+            position.x = (value._startPos.coord.x + value._endPos.coord.x) / 2;
+            position.y = (value._startPos.coord.y + value._endPos.coord.y) / 2;
+            if (
+                position.y < this.bound._min._lat - 0.01
```
`0.01`을 상수처리

senxation commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,307 @@
+        this.setState({ showModal: !showModal });
+    };
+
+    _toggle7 = () => {
```
이것만 underscore로 시작하는 이유가 모에요?
그리고 네이밍으로 무슨 일을 하는 메서드인지 알아채기 힘드네요.

JEONGJIHUN replied:
내부함수로 사용하려고 만들었지만 사용 방향이 변해 수정 완료(myInfoToggle)했으나 아직 pr을 보내지 못해 수정 전 함수입니다.

senxation replied:
pr을 보낸다는건 feature 작업중인거를 develop 머지한다는거죠?

senxation commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,307 @@
+    _toggle7 = () => {
+        const { check7 } = this.state;
+        this.setState({ check7: !check7 });
+        return check7;
```
리턴값이 기존값인게 이상한데, 살펴보니 리턴값 받아서 쓰는데도 없는거 같아요.

JEONGJIHUN replied:
이것도 수정 완료(myInfoToggle)했으나 아직 pr을 보내지 못해 수정 전 함수입니다.

flyingSquirrel-dev committed:
modfiy and fix bugs about drawingMode

senxation commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,307 @@
+                factors.push(key);
+            }
+        });
+        axios
```
이후 코드가 `mainPageLoad`랑 중복되는 부분이 많은데 일반화 해야겠네요.

senxation commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,307 @@
+                            }
+                        />
+                    ) : null}
+                    <div style={{ display: !showFilter ? 'block' : 'none' }}>
```
style을 inline으로 직접 지정하는 것 보다는 style은 `less(css)`에 정의하고 class명을 toggle 하는 방식을 쓰는게 좋아요.

JEONGJIHUN replied:
넵 수정했습니다~!

senxation commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,307 @@
+                            factorLoad={this.factorLoad}
+                        />
+                    </div>
+                    <div style={{ display: showDraw ? 'block' : 'none' }}>
```
여기도 마찬가지

JEONGJIHUN replied:
넵 수정했습니다.

senxation commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,307 @@
+                    </div>
+                    <div style={{ display: showDraw ? 'block' : 'none' }}>
+                        <DrawContainer
+                            closeFn={this.showDraw}
```
`closeFn` 이렇게 postfix로 네이밍하는 것도 다른 함수들과는 다른 방식으로 처리되었네요.
통일성을 위해 그냥 떼는게 좋겠음.

`onClose` 정도나, 혹은 아래쪽엔 `handle~` 이런식으로 썼으니 `handleToggle` 뭐 이런식이어도 좋겠고요.

flyingSquirrel-dev replied:
반영했습니다!

senxation commented on packages/client/src/Components/DrawContainer.js:
```diff
@@ -0,0 +1,42 @@
+                <div id="tabMenu">
+                    <div
+                        className="eachTabMenu"
+                        style={{
```
less 로 빼내기

flyingSquirrel-dev replied:
반영했습니다!

senxation commented on packages/client/src/Components/DrawContainer.js:
```diff
@@ -0,0 +1,42 @@
+                            color: '#fff'
+                        }}
+                    >
+                        {`그리기`}
```
{``} 로 감쌀 필요가 없음

flyingSquirrel-dev replied:
반영했습니다!

senxation commented on packages/client/src/Components/Drawing.js:
```diff
@@ -0,0 +1,213 @@
+                    console.log(selectedButton === shape);
+
+                    return (
+                        <Button
```
key 정의 안되었음

senxation commented on packages/client/src/Components/Drawing.js:
```diff
@@ -0,0 +1,213 @@
+    state = {
+        index: 0,
+        theNumberOfFigure: [],
+        shapes: ['line', 'arrow', 'square', 'circle', 'polygon'],
```
상수로 뺄것.
고정된 값이라면 state로 정의할 필요없음.

senxation commented on packages/client/src/Components/Drawing.js:
```diff
@@ -0,0 +1,213 @@
+        const naver = window.naver;
+        const { map } = this.props;
+        const icons = ['line', 'arrow', 'square', 'circle', 'polygon'];
+        const overlays = [Line, Circle, Rect, Circle, Circle]; // Change name of index to actual overlay name of import
```
icons에 대응하는 overlay 의 형태인거 같은데 상수로 빼서
`const SHAPES = [{ icons: 'line', overlay: Line }, ...]` 이런식으로 정의하는게 좋겠음.

senxation commented on packages/client/src/Components/Drawing.js:
```diff
@@ -0,0 +1,213 @@
+
+        for (let index = 0; index < icons.length; index++) {
+            if (selectedIcon === icons[index]) {
+                Shape = overlays[index];
```
break; 안하면 배열 계속 순회함.

senxation commented on packages/client/src/Components/Filter.js:
```diff
@@ -0,0 +1,61 @@
+
+    render() {
+        const { check7, factorLoad, _toggle7 } = this.props;
+        const factorBox = [
```
상수...

senxation commented on packages/client/src/Components/Filter.js:
```diff
@@ -0,0 +1,61 @@
+                        onKeyPress={_toggle7}
+                        role="button"
+                        tabIndex="0"
+                        style={{
```
css로 빼내기

JEONGJIHUN replied:
수정했습니다~!

senxation commented on packages/client/src/Components/Filter.js:
```diff
@@ -0,0 +1,61 @@
+                            backgroundColor: check7 ? '#4d55b2' : '#fff'
+                        }}
+                    >
+                        {`내 정보 보기`}
```
감쌀필요 없음

flyingSquirrel-dev replied:
@senxation 감싸지 않으면 react/jsx-one-expression-per-line 룰에 걸리는데, 저희가 .eslintrc에 규칙을 추가해도 될까요?

senxation replied:
@cloiepark 엥? 저는 에러 안나는거 같은데. 환경설정이 다른가

flyingSquirrel-dev replied:
@JEONGJIHUN @DoheeKang @vjbhang 다른 분들도 진행하실 때 백틱빼고 하면 lint에서 error 보였던 것 같은데 맞는지요?
/client/src/.eslintrc에 "react/jsx-one-expression-per-line": 0 추가해야하는 건지 고민이네요...ㅠ

senxation commented on packages/client/src/Components/FilterBox.js:
```diff
@@ -0,0 +1,31 @@
+                role="button"
+                tabIndex="0"
+            >
+                #
```
```suggestion
                {`# ${factor}`}
```

JEONGJIHUN replied:
수정했습니다~!

senxation commented on packages/client/src/Components/MyDrawings.js:
```diff
@@ -0,0 +1,88 @@
+                    {fakeData.map(figure => {
+                        return (
+                            <div>
+                                <li
```
`key`가 상위 div에 있어야할것 같음

flyingSquirrel-dev replied:
반영했습니다!

senxation commented on packages/client/src/Components/Toolbox.js:
```diff
@@ -0,0 +1,113 @@
+                        onClick={() => {
+                            this.handleOnFilter(true);
+                        }}
+                        onKeyPress={() => { }}
```
불필요하면 삭제

flyingSquirrel-dev replied:
Toolbox.js 삭제했습니다!

senxation commented on packages/client/src/CustomOverlay/Circle.js:
```diff
@@ -0,0 +1,53 @@
+// import * as d3 from 'd3';
+import Shape from './Shape';
+
+var Circle = function(options) {
```
Circle 이 d3로 그린게 아니고 `window.naver.maps.OverlayView` 로 그린거네요.
이렇게 하면 드래그 하는 동안 그리는 과정을 그려내지는 못할 것 같은데, 그 과정은 지금 진행 중이겠죠??

senxation commented on packages/client/src/Module/shapeButtonStates.js:
```diff
@@ -0,0 +1,80 @@
+        this.setState({ polygonButtonToggle: false });
+    } else {
+        this.setState({ lineButtonToggle: false }, () => console.log('lineButtonWhenTrue: ', lineButtonToggle));
+    }
```
```suggestion
    }
    
    this.setState({ 
            lineButtonToggle: false,
            arrowButtonToggle: arrowButtonToggle === false ? true : false,
            squareButtonToggle: false,
            polygonButtonToggle: false,
            circleButtonToggle: false 
        });
```

senxation commented on packages/client/src/Module/shapeButtonStates.js:
```diff
@@ -0,0 +1,80 @@
+
```
이 파일 자체를 분리할 필요까진 없는거 같은데.
그리고 실제로 쓰는데도 없는거 같은데요?

vjbhang replied:
삭제해야합니다!

senxation commented on packages/client/src/Module/Button.js:
```diff
@@ -0,0 +1,71 @@
+/* eslint-disable no-nested-ternary */
```
Module 폴더보다 Component 폴더에 두는게 적절할듯

vjbhang replied:
참고하겠습니다

senxation commented on packages/server/routes/userRouter/controller.js:
```diff
@@ -0,0 +1,210 @@
+                    factor_id: { [Op.in]: factorIdArray },
+                    center_lat: {
+                        [Op.between]: [
+                            bound._min._lat - 0.01,
```
클라이언트에서도 마진값을 추가한거 같은데 여기서도 하네요.
클라이언트에서는 안하도록 빼는게 낫겠음.

senxation commented on packages/server/routes/userRouter/controller.js:
```diff
@@ -0,0 +1,210 @@
+                            bound._min._lng - 0.01,
+                            bound._max._lng + 0.01
+                        ]
+                    }
```
결국 중점만으로 쿼리하는거죠?

Commit:
Draw Polygon
* using line class

Commit:
Merge branch 'dev' of https://github.com/hogangnono/codestates into drawing

flyingSquirrel-dev committed:
change the default value of this.state.name from jihye to undefined

flyingSquirrel-dev committed:
change function name of closeFn to handleToggle

flyingSquirrel-dev committed:
modify details according to comments

JEONGJIHUN committed:
Modified function name; reflected code review almost

JEONGJIHUN committed:
Merge pull request #63 from cloiepark/feature/drawingMode

Feature/drawing mode and Handling Login

JEONGJIHUN committed:
Merged cloie code and my Modified code

Commit:
Modify draw polygon
* Closed path

Commit:
Make Arrow interactively
* Child of Line
* Make triangle and add to line using 'marker-end'

Commit:
Refactoring Drawing.js
* Combine line and rect logic

JEONGJIHUN committed:
ejected load api function; made api.js and  constant.js;

JEONGJIHUN committed:
Modified api to constant

JEONGJIHUN committed:
Merge branch 'dev' into drawing

JEONGJIHUN committed:
Merge pull request #64 from DoheeKang/drawing

Draw polygon and arrow

flyingSquirrel-dev committed:
add tip Modal to let users know how to change Drawing/Filter mdoe

flyingSquirrel-dev committed:
Merge branch 'dev' of https://github.com/hogangnono/codestates into feature/floatingTipBox

flyingSquirrel-dev committed:
modify MyDrawingElements.js to show the list of drawings

DoheeKang committed:
Modify page title and logo

DoheeKang committed:
Delete blue line when button focus

JEONGJIHUN committed:
Merged sally code

JEONGJIHUN committed:
Merge pull request #65 from JEONGJIHUN/feature/filterboxcomponent

Feature/filterboxcomponent

JEONGJIHUN committed:
Merge branch 'dev' into feature/floatingTipBox

JEONGJIHUN committed:
Merge pull request #66 from cloiepark/feature/floatingTipBox

(Feature) Floating tip box and the list of drawings

JEONGJIHUN committed:
Merge branch 'dev' into dev

JEONGJIHUN committed:
Merge pull request #67 from DoheeKang/dev

Delete blue line on button

flyingSquirrel-dev committed:
pull from upstream dev

DoheeKang committed:
Resolve conflict when pull upstream dev

flyingSquirrel-dev committed:
modify loading image for login feature and modify css

JEONGJIHUN committed:
got nearbyData using load api; modified loadHandle.js; modified nearbyfactor less files

JEONGJIHUN committed:
Modified load api; got nearbydata for user;making post slack noti; use transaction;

flyingSquirrel-dev committed:
modify save feature

flyingSquirrel-dev committed:
modify save feature

flyingSquirrel-dev committed:
remove console.log in /user/save Router

JEONGJIHUN committed:
Merge pull request #68 from cloiepark/feature/save

Feature/save

JEONGJIHUN committed:
Merge branch 'dev' into CalRatio

DoheeKang committed:
calculate rect and circle ratio when zoom in out
* line data: client -> coord, offset

DoheeKang committed:
Merge branch 'CalRatio' of https://github.com/DoheeKang/codestates into CalRatio

JEONGJIHUN committed:
Merged cloie code

JEONGJIHUN committed:
Merge pull request #70 from JEONGJIHUN/feature/nearbyFactor

Feature/nearby factor

JEONGJIHUN committed:
Merge branch 'dev' into CalRatio

JEONGJIHUN committed:
Merge pull request #71 from DoheeKang/CalRatio

Cal ratio of rect and circle

DoheeKang committed:
Draw line interactivly when draw left top to right botton
* make svg with mouse interact
* modify start point of line

flyingSquirrel-dev committed:
change API key and resolve lint errors

flyingSquirrel-dev committed:
delete center lat, lng column and add start/end lat/lng column

DoheeKang committed:
Calculate Line ratio when zoom in out

DoheeKang committed:
Calculate arrow and polygon ratio

DoheeKang committed:
Pull and Merge with origin CalRatio

DoheeKang committed:
Merge branch 'CalRatio' of https://github.com/DoheeKang/codestates into CalRatio

DoheeKang committed:
Delete consolelog

JEONGJIHUN committed:
Merge pull request #72 from cloiepark/modify/dbSchema

Modify/db schema

DoheeKang committed:
Clean up code with rect and circle

flyingSquirrel-dev committed:
change datatype of figures in Figure table from STRING to TEXT because of limit of string length

JEONGJIHUN committed:
making web hook and merged code

JEONGJIHUN committed:
Merge pull request #74 from cloiepark/modify/dbSchema

feature/save: modify DB schema and client logic

JEONGJIHUN committed:
export post slack api function and added slack api path in constants.js

JEONGJIHUN committed:
Merged cloie code

DoheeKang committed:
Modify error about rect and circle
* when draw rect bigger, it dissaper
* clean up code

JEONGJIHUN committed:
Merge pull request #75 from DoheeKang/CalRatio

Cal ratio about all shape

DoheeKang committed:
Fix error about circle and rect
* when user click right button while drawing circle or rect,
it can't finish drawing

JEONGJIHUN committed:
Merged sally code and modified saveHandle.js

DoheeKang committed:
Modify error that when draw line, list generate only once
* delete generate function when just generate line point

DoheeKang committed:
Maintain nearBylist when filtering factor
* use other value for entire factor list and filtering factor list

JEONGJIHUN committed:
Changed this.newToggleBox in App.js; deleted factorBox in Filter.js; renamed NearByFactorItems elements in NearbyFactorDialog.js;

JEONGJIHUN committed:
Modified index.html

vjbhang committed:
Fixed Button Toggle for new Shape

- One line was missing

JEONGJIHUN committed:
Deleted useless code

JEONGJIHUN committed:
Merge pull request #78 from DoheeKang/modifyError

Modify error

JEONGJIHUN committed:
Merge pull request #80 from vjbhang/buttonFixDev

Fixed Button Toggle for new Shape

JEONGJIHUN committed:
Merge branch 'dev' into feature/webHookAndModifiedCode

JEONGJIHUN committed:
Merge pull request #81 from JEONGJIHUN/feature/webHookAndModifiedCode

Feature/web hook and modified code

flyingSquirrel-dev committed:
feature/deleteShape

vjbhang committed:
editted Close Button for showDraw method in App.js

- Unstaged shapes will be removed!

flyingSquirrel-dev committed:
change DB schema

flyingSquirrel-dev committed:
complete delete feature

flyingSquirrel-dev committed:
change active mode between 필터 and 그리기

DoheeKang committed:
when user click right button, that point is deleted
* deleted last point in the line Data
* line, arrow, polygon

DoheeKang committed:
Change arrow head position
* from end to start
* reverse head direction

DoheeKang committed:
Make logic that decide color and fill or not

JEONGJIHUN committed:
Made slack alert and confirm/refuse button functon

DoheeKang committed:
Modify error about fill or not
* if don't want fill, the value is 'none'

JEONGJIHUN committed:
Modified SLACK_GENERAL_PATH because I use ngrok

JEONGJIHUN committed:
Merge pull request #83 from cloiepark/DB/addDBcoulum

(DB) add new columns

JEONGJIHUN committed:
Merge branch 'dev' into removeStagedShapesDev

JEONGJIHUN committed:
Merge pull request #84 from vjbhang/removeStagedShapesDev

editted Close Button for showDraw method in App.js

JEONGJIHUN committed:
Merge pull request #85 from cloiepark/feature/deleteShapes

Feature/deleteShapes and activeMode(필터,그리기)

JEONGJIHUN committed:
Merged team code

DoheeKang committed:
Show factor button when shape is clicked

JEONGJIHUN committed:
Merge pull request #86 from JEONGJIHUN/feature/slackWebHook

Feature/slack web hook

DoheeKang committed:
Css the factor box and modify error about factor box
* if user didn't choose factor or figure option,
user can't draw figure

DoheeKang committed:
Modify error about arrow and polygon
* when user right click just after the first left click,
delete that figure

flyingSquirrel-dev committed:
modify DrawingList

flyingSquirrel-dev committed:
delete unuseless code in Drawing.js

DoheeKang committed:
Be variable that spare place
* store that value in 'spare'
* width and height is bigger 2px

DoheeKang committed:
Change state 'inInshapeCreateMode' false, when finish drawing

JEONGJIHUN committed:
Merge pull request #87 from cloiepark/feature/DrawingList

feature/DrawingList

JEONGJIHUN committed:
Merge branch 'dev' into modifyDrawing

JEONGJIHUN committed:
Merge pull request #88 from DoheeKang/modifyDrawing

Modify drawing error and detail

DoheeKang committed:
Modify save and load
* Now shape will not draw, if there are no color and fill

JEONGJIHUN committed:
test puppeteer and url

JEONGJIHUN committed:
Merge pull request #90 from DoheeKang/detail

Modify save and load

JEONGJIHUN committed:
Merge branch 'dev' of https://github.com/hogangnono/codestates into feature/CaptureImage

JEONGJIHUN committed:
added less and connected ec2; studying url and react build

flyingSquirrel-dev committed:
modify to get nearby list well

JEONGJIHUN committed:
Merge pull request #91 from JEONGJIHUN/feature/CaptureImage

Feature/capture image

JEONGJIHUN committed:
Merge pull request #92 from cloiepark/feature/NearbyDrawingSet

feature/nearbylist

DoheeKang committed:
Modify error that part of line dissapear when load DB
* also modify error that zoom in out
* spare is calculate accordingly ratio

JEONGJIHUN committed:
Merge pull request #93 from DoheeKang/detail

Modify error that part of line dissapear when load DB

JEONGJIHUN committed:
Modified css and switch in loadHandle.js

DoheeKang committed:
Delete figure when they are out of bound
* make a spare space that +- 0.01 of lat, lng

vjbhang committed:
Added styling; Add title/ value (description)

- Added styling for pop-up modal
- Added description
- Added title and description state.
- Will connect with saveHandle!

vjbhang committed:
remove unnecessary (comments)

DoheeKang committed:
Delete when figure is out of screen

vjbhang committed:
updated Savehandle so that ir receives actual data

- Updated Description Bar
- It works together with save handle method

vjbhang committed:
Handled edge cases for removing description modal

- Handle edge cases where the user doesn't submit description or follows
different procedure instead of desired.

vjbhang committed:
Refined Styling

flyingSquirrel-dev committed:
modify css

JEONGJIHUN committed:
Merge pull request #94 from JEONGJIHUN/feature/drawingToolCss

Modified css and switch in loadHandle.js

JEONGJIHUN committed:
Modified filter list and drawing list; made legend list;

JEONGJIHUN committed:
Merge branch 'dev' into detail

JEONGJIHUN committed:
Merge pull request #95 from DoheeKang/detail

delete figure that out of screen

JEONGJIHUN committed:
Merge branch 'dev' into descriptionBarDev!

JEONGJIHUN committed:
Merge pull request #97 from vjbhang/descriptionBarDev!

Added Description Modal

JEONGJIHUN committed:
Modified description modal design and merged vincent,sally code

flyingSquirrel-dev committed:
modify my,drawing,mydrawingElement etc CSS and create drawingSetTitleDescription Modal

flyingSquirrel-dev committed:
pull upstream dev and resolve conflicts

flyingSquirrel-dev committed:
resolve warning and error when starting React

flyingSquirrel-dev committed:
modify bugs when saving

flyingSquirrel-dev committed:
Merge pull request #98 from JEONGJIHUN/feature/drawingToolCss

Feature/drawing tool css

flyingSquirrel-dev committed:
pull upstream dev

flyingSquirrel-dev committed:
add key value to resolve warning

JEONGJIHUN committed:
Making intro page;

DoheeKang committed:
Change mouse when it is draw mode or not
* when drawing mode it is crosshair
* and not, it is grab

DoheeKang committed:
modify rect error that draw a little bit left top

DoheeKang committed:
Merge branch 'detail' of https://github.com/DoheeKang/codestates into detail

DoheeKang committed:
eslient error

JEONGJIHUN committed:
Merge pull request #99 from cloiepark/feature/nearByList

Feature/My, Drawing

JEONGJIHUN committed:
Merge pull request #100 from JEONGJIHUN/feature/drawingToolCss

Making intro page

DoheeKang committed:
Merge branch 'dev' of https://github.com/hogangnono/codestates into detail

DoheeKang committed:
Modify error that shape box apper only once
* draw is finish, showDrawBox make false

DoheeKang committed:
Modify error about cursor change

DoheeKang committed:
Delete other info when user drawing
* after drawing, the drawing shown

JEONGJIHUN committed:
Merge pull request #102 from DoheeKang/detail

change cursor and error about shape box

JEONGJIHUN committed:
Modified css;

flyingSquirrel-dev committed:
modify NearbyList.js and CSS

flyingSquirrel-dev committed:
set utf8 to Drawing table

JEONGJIHUN committed:
Merge pull request #104 from cloiepark/feature/nearByList

Feature/near by list

JEONGJIHUN committed:
Fixed save logic

JEONGJIHUN committed:
Merge pull request #105 from JEONGJIHUN/feature/IntopageAndFixedBugs

Feature/intopage and fixed bugs

davidshimjs commented on packages/server/routes/userRouter/index.js:
```diff
@@ -0,0 +1,9 @@
+const router = require('express').Router();
+const controller = require('./controller');
+
+router.post('/', controller.signup);
```
RESTful API 설계 방식이 있는데 그것을 완전히 따르진 않더라도 Method(GET, POST, PUT, DELETE)와 URI가 리더블한 API가 좋아요.
- POST /user
- GET /user/factors
- POST /user/factors
- DELETE /user/factors

davidshimjs commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,506 @@
+
+    componentDidMount = async () => {
+        const naver = window.naver;
+        const map = await new naver.maps.Map(d3.select('#map').node(), {
```
`document.querySelector('#map')` 으로 썼으면 네이티브 DOM API를 잘 알고 있는 것 같아서 좋았겠네요 ㅎㅎ caniuse.com 을 참고해서 두루두루 사용 가능한건 네이티브로 쓰면 좋아요. 속도도 빠르고 의존성도 줄이고.

davidshimjs commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,506 @@
+
+        this.setState({ map });
+        this.bound = map.getBounds();
+        this.mainPageLoad(map);
```
작명이 중요한데 이럴 경우 메인 페이지를 로드하고 호출하는 건지, 메인 페이지를 로드하는건지 좀 헷갈리네요. 이름만으로 행위를 짐작하게 작성하는게 중요한 것 같아요. 내용 보니깐 메인 페이지를 로드하는게 아니라 준비된 데이터를 단순히 그리기만 하네요. 그 [아래](https://github.com/hogangnono/codestates/pull/62/files#diff-0821fde1582d716e8797cf23613f5667R277)에 유사한 행위가 있어서 중복되는 단위를 쪼개면 좋을 것 같아요.

davidshimjs commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,506 @@
+    mainPageLoad = map => {
+        const { name, factors } = this.state;
+        const bound = this.bound;
+        // this.setState는 Promise가 아닌데 async await 안써도 되지 않나요? (from Cloie)
```
요런 리뷰를 남길때 https://reactjs.org/docs/react-component.html#setstate 근거를 남기면 더 설득력 있고 동료의 시간을 줄여줄 수 있죠

davidshimjs commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,506 @@
+    }
+
+    changeDrawingSetDescription = text => {
+        this.setState({ drawingSetDescription: text });
```
이 클래스에서 setState 별로 메서드를 따는건 좀 불필요한 행위 같긴 하네요. 없어도 state의 key 이름을 잘 지으면 충분히 읽을 수 있을 것 같아요.

davidshimjs commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,506 @@
+            });
+            this.isDelete = true;
+        } else {
+            if (this.isDelete) {
```
불필요한 블럭
`} else if (this.isDelete) {`

davidshimjs commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,506 @@
+        });
+    };
+
+    isDelete = false;
```
다른건 생성자에서 초기화 했는데 왜 얘는 여기 있어요?

davidshimjs commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,506 @@
+            activeFilter: 'active',
+            activeDraw: '',
+            MyInfoButton: false,
+            showDraw: false,
```
자바스크립트가 타입이 정해져있지 않기 때문에 네이밍할 때 변수에 동사를 먼저 쓰는게 좀 꺼려지긴 해요.

davidshimjs commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,506 @@
+    toggleAllDraw = () => {
+        const { showDraw, map } = this.state;
+        if (showDraw) {
+            Object.entries(this.drawList).forEach(([key, value]) => {
```
아래 코드도 그렇고 이름은 list로 지어놓고 map을 쓰고 있어서 매번 불필요한 형변환이 필요하네요. (이런게 있었구나...) delete 때문에 그런거 같은데 Array.splice나 filter 등을 쓰던지 해야할 것 같아요. 의미에 맞게.. 나중에 성능 등의 문제가 될 수 있어요. length 도 없고요.

davidshimjs commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,506 @@
+                value.setMap(null);
+                delete this.drawList[key];
+            });
+            this.isDelete = true;
```
위에꺼 list로 관리했으면 .length로 판별해도 될텐데..

davidshimjs commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,506 @@
+            endPos.y = value._lineData[value._lineData.length - 1].coord.y;
+
+            if (
+                (startPos.x < this.bound._min._lng - 0.01
```
매직 넘버. 0.01이 의미가 있는 값이면 의미를 적어서 상수로 사용하는게 좋아요. 위경도는 그리고 타원 좌표라 요렇게 계산하면 정확하지 않아요. 직교 좌표로 변환하거나 네이버에서 제공하는 API를 사용해서 지도 안에 들어왔는지를 확인해야 해요.

davidshimjs replied:
그리고 보통 이름 앞이나 뒤에 `_`가 붙어있으면 private 하게 쓰고 싶다는거니 꺼내 쓰지 않는게 좋아요. 자바스크립트는 제한자가 없어서 네이밍 컨벤션으로 다루는게 많아요. 이런건 다른 오픈소스 코드를 많이 읽어보세요.

davidshimjs commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,506 @@
+
+    showFilter = () => {
+        const { showFilter, activeFilter } = this.state;
+        if (activeFilter === 'active') {
```
이런 값도 active가 들어갈지 뭐가 들어갈지 모르니 값의 목록을 상수로 빼는게 커뮤니케이션 하기 좋아요.

davidshimjs commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,506 @@
+    showDraw = () => {
+        const { showDraw, drawingData, activeDraw } = this.state;
+        if (drawingData.length) {
+            const pressedConfirm = confirm(
```
```
if (window.confirm('...')) {
} else {
}
```
이렇게 쓰는게 더 간결하고 어렵지도 않아서 이런건 보통 따로 변수로 빼지 않아요. 조건문으로 변수로 뺄때는 조건문 자체만으로 이해가 어려울 때인 것 같네요. 주석 대신으로 변수명을 줘요. ex) a && b && (c || d) ..

davidshimjs commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,506 @@
+                '저장하지 않고 그리기 창을 닫으면 그린 정보는 모두 사라집니다!\n그래도 그리기 창을 닫으시겠어요?'
+            );
+            if (pressedConfirm) {
+                for (let index = 0; index < drawingData.length; index++) {
```
forEach 잘 쓰다 여기는 for문을 썼네요. 표준 문법은 버전 일관성도 중요한 것 같아요. 여러 버전이 한 문서에 있으면 읽기가 어렵거든요. (자주 섞이긴 하지만...)

davidshimjs commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,506 @@
+                    drawingData[index].figure.onRemove();
+                }
+                this.setState({ drawingData: [] });
+                this.setState({ descriptionModalState: false });
```
setState를 끊어서 하면 이벤트도 나눠서 발생되고, 가능하면 묶어서 하는게 읽기도 좋아요.
```
this.setState({
    drawingData: [],
    descriptionModalState: false
});
```

davidshimjs commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,506 @@
+                }
+                this.setState({ drawingData: [] });
+                this.setState({ descriptionModalState: false });
+            } else if (!pressedConfirm) {
```
의미 없는 조건문이고 2 depth 들어가서 return 하면 복잡도가 너무 높아서 실수할 가능성이 높아요. 가급적 1 depth로 빼서 return 하는걸 추천합니다.

```
if (drawingData.length) {
    if (!confirm()) {
        return;
    }

    forEach...
    setState..

    if (activeDraw === ...) {
    ...
}
```

davidshimjs commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,506 @@
+    myInfoToggle = () => {
+        const { MyInfoButton } = this.state;
+        this.setState({ MyInfoButton: !MyInfoButton });
+        this.factorLoad(undefined, !MyInfoButton);
```
빈 값을 넘긴다는 의미의 null이 더 적합할 것 같네요. 그리고 앞이 비어있는 경우가 더 많다면 설계할 때 순서를 잘못하지 않았는지 의심해볼 필요가 있어요.

davidshimjs commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,506 @@
+        let nearbyData;
+
+        // 기존 지도에 있던 정보를 지워줌
+        Object.entries(this.drawList).forEach(([key, value]) => {
```
반복되는 행위이므로 메소드를 따로 빼주면 좋겠네요

davidshimjs commented on packages/client/src/App.js:
```diff
@@ -0,0 +1,506 @@
+            delete this.drawList[key];
+        });
+        // 유저 호재 보기
+        if (toggle) {
```
newToggleBox 맵에 key는 카테고리 value는 활성화 상태(Boolean)을 저장해놓고 toggle = true로 보내면 false로 리셋하고 category 를 키는거고, toggle = false 면 현재 상태에서 category에 해당하는 것만 토글해서 호출한다는 뜻.... 맞죠?

```
loadFactors = (category, resetAll = false) => {
...
  Object.entries(this.newToggleBox).forEach([key, value]) => {
    let isActivated = resetAll ? false : value;

    if (key === category) {
      toggleObj[key] = !isActivated;
    }
  });
}
```
요러면 루프 한번에 돌 수 있을 것 같네요. 그리고 애초에 자료형을 잘못 잡아서 보낼 때 형변환을 해야하는게 문제 같아요. 켜진 애들만 list로 관리하고, 어떤 애들이 있는지 목록을 따로 관리하면 좋을 것 같아요. list로 관리하면 resetAll할 때 다 날리면 되잖아요.

```
if (resetAll) {
    toggleList = [];
}

toggleList.push(category);
```

davidshimjs reviewed (commented):
리뷰를 한다고 잡았는데 다 못해드려서 죄송하네요ㅠㅠ 고생하셨습니다~! (보통 이럴 땐 리뷰할 준비가 안된 것 같다며 정리해서 PR을 다시 올리라고 해요..)

추가로
- boilerplate를 사용한 것 같은데 이 프로젝트에서 쓰지 않는 코드가 많이 남아있는게 좀 아쉬웠고 (해당 내용은 인지하고 있지 않은 것 같음)
- 주석처리한 코드들도 날려놓거나, 왜 주석처리했는지 적어놔야 할 것 같아요.
- 디버깅 코드 같은건 develop에 머지되기 전에 없어져야 합니다.

flyingSquirrel-dev committed:
modify Server/load Router and show shpaes title

JEONGJIHUN committed:
Merge pull request #106 from cloiepark/feature/nearByList

modify load Router and nearByFactorItems

flyingSquirrel-dev committed:
modify nearByList. load data per drawing set

flyingSquirrel-dev committed:
Merge branch 'dev' of https://github.com/hogangnono/codestates into feature/nearByList

DoheeKang committed:
Load user data list
* load data in the LoginModal
* fix it in App.js

flyingSquirrel-dev committed:
Merge pull request #107 from cloiepark/feature/nearByList

feature/nearByList

flyingSquirrel-dev committed:
Merge pull request #108 from DoheeKang/detail

Load user data list

flyingSquirrel-dev committed:
remove useless comments

JEONGJIHUN committed:
Fixed errors

JEONGJIHUN committed:
Merged sally and cloie code

JEONGJIHUN committed:
Merge pull request #109 from JEONGJIHUN/feature/IntopageAndFixedBugs

Feature/intopage and fixed bugs

flyingSquirrel-dev committed:
modify My

flyingSquirrel-dev committed:
Merge branch 'dev' of https://github.com/hogangnono/codestates into feature/my

flyingSquirrel-dev committed:
modfiy MyDrawings and create new Router named userdrawing to get drawing set info

flyingSquirrel-dev committed:
change API host to EC2 address

DoheeKang committed:
Save factor id

JEONGJIHUN committed:
Merge pull request #110 from cloiepark/feature/my

Feature/my

JEONGJIHUN committed:
Merge pull request #111 from DoheeKang/detail

Save factor id

JEONGJIHUN committed:
Deleted useless console.log

JEONGJIHUN committed:
Merge branch 'dev' of https://github.com/hogangnono/codestates into dev

flyingSquirrel-dev committed:
modify MyDrawingElement to show drawingTitle

flyingSquirrel-dev committed:
modify myDrawingElement to show shape name when title is not given

flyingSquirrel-dev committed:
fix bug that color does not match with color of factor user selected

flyingSquirrel-dev committed:
apply overflow auto

flyingSquirrel-dev committed:
modify css

DoheeKang committed:
Modify error that load data when draw mode

JEONGJIHUN committed:
Merge pull request #114 from DoheeKang/detail

Modify error that load data when draw mode

JEONGJIHUN committed:
Modified drawing new toggle box index key

flyingSquirrel-dev committed:
Merge branch 'dev' of https://github.com/hogangnono/codestates into dev

flyingSquirrel-dev committed:
modify things

flyingSquirrel-dev committed:
modify map key

JEONGJIHUN committed:
Modified Draw in App.js

JEONGJIHUN committed:
Added config file in gitignore

JEONGJIHUN committed:
Added DS_Store file in gitignore

JEONGJIHUN committed:
Edited Readme.md

JEONGJIHUN committed:
Added gif in readme.md

flyingSquirrel-dev committed:
fix right click bugs after building

JEONGJIHUN committed:
Merge pull request #115 from cloiepark/build

Build 후 우클릭으로 도형 종료 안되는 이슈 수정

JEONGJIHUN committed:
Merge branch 'dev' of https://github.com/hogangnono/codestates into dev

JEONGJIHUN committed:
Changed Client id

JEONGJIHUN committed:
Edited this.factorLoad in myInfoToggle function

JEONGJIHUN committed:
Changed server address

JEONGJIHUN committed:
Updated readme.md

JEONGJIHUN committed:
i

flyingSquirrel-dev committed:
fix a toggle btn in Drawing.js

flyingSquirrel-dev committed:
bugfix on drawingComponentContainer

flyingSquirrel-dev committed:
bug fix on saving factor_id

JEONGJIHUN committed:
Updated README.md

JEONGJIHUN committed:
Merge pull request #116 from cloiepark/feature/modifyToggleBtn

도형 토글 버튼 버그 수정

JEONGJIHUN committed:
Merge branch 'dev' of https://github.com/hogangnono/codestates into dev

flyingSquirrel-dev committed:
add img files

flyingSquirrel-dev committed:
add img files to modify README.md

flyingSquirrel-dev committed:
modify README

flyingSquirrel-dev committed:
add a img file

JEONGJIHUN committed:
Merge pull request #117 from cloiepark/dev

README.md 수정

JEONGJIHUN committed:
Merge branch 'dev' into dev

JEONGJIHUN committed:
Merge pull request #118 from JEONGJIHUN/dev

Updated Readme.md
